### PR TITLE
Drop support for Python-2.7 in the client.

### LINKED ISSUE #4614: Drop Python 2.7 support in Python client
The Python client supports Python 2.7 as well as recent Python 3 versions:

https://github.com/taskcluster/taskcluster/blob/cf14539f4cd5ba92e293eb456291e1897c2eb41c/clients/client-py/setup.py#L96-L99

We have a similar test matrix that includes later Python 3 versions:

https://github.com/taskcluster/taskcluster/blob/cf14539f4cd5ba92e293eb456291e1897c2eb41c/clients/client-py/tox.ini#L2-L7

Many projects have dropped Python 2.7 support after it reached end-of-line on Jan 1, 2020. Taskcluster should drop it as well, or plan to in a future release.

The documentation implies that the synchronous code is for Python 2 and the async code is for Python 3. The sync code should continue to work in Python 3, and I think deciding about future support of synchronous code should be separate from dropping Python 2.7.

### PR DESCRIPTION
**Pull request by djmitche**
Github Bug/Issue: Fixes #4614 

Always fun to delete some code :)

### CODE REVIEW

File: clients/client-py/README.md [hunk 1]
```diff
@@ -22,9 +21,8 @@ required, and it's unusual to configure any other options aside from
 `credentials`.
 
 For each service, there are sync and async variants.  The classes under
-`taskcluster` (e.g., `taskcluster.Queue`) are Python-2 compatible and operate
-synchronously.  The classes under `taskcluster.aio` (e.g.,
-`taskcluster.aio.Queue`) require Python >= 3.6.
```
Comment (grahamalama) [inline clients/client-py/README.md:27]: > require Python >= 3.6
Is this still true?

File: clients/client-py/setup.py [hunk 2]
```diff
@@ -32,11 +31,10 @@
     'mohawk>=0.3.4',
     'slugid>=2',
     'taskcluster-urls>=12.1.0',
-    'six>=1.10.0',
 
     # python-3 only
```
Comment (grahamalama) [inline clients/client-py/setup.py:35]: Probably don't need this heading anymore, right?

File: clients/client-py/setup.py [hunk 3]
```diff
@@ -96,8 +90,10 @@ def run_tests(self):
         },
         cmdclass={'test': Tox},
         zip_safe=False,
-        classifiers=['Programming Language :: Python :: 2.7',
-                     'Programming Language :: Python :: 3.5',
-                     'Programming Language :: Python :: 3.6',
-                     'Programming Language :: Python :: 3.7'],
+        classifiers=[
```
Comment (grahamalama) [inline clients/client-py/setup.py:93]: Two questions here:
- is this client Python 3.10 compatible? (or is that out of scope to account for in this PR?)
- I noticed in other places that we removed references to Python 3.6 (which is also at EOL). Should we remove the classifier here?

File: clients/client-py/tox.ini [hunk 4]
```diff
@@ -1,6 +1,5 @@
 [tox]
 envlist =
-    py27
     py36
```
Comment (grahamalama) [inline clients/client-py/tox.ini:3]: Similar question to above about Python 3.6 & 3.10

**Review by grahamalama — COMMENTED** [review]
Love cutting lines of code. Just a few questions left inline.

File: clients/client-py/setup.py (hunk 3 above)
Comment (djmitche) [reply clients/client-py/setup.py:91]: I think that's out of scope.  We typically haven't dropped support for old versions until they become a nuisance (as 2.7 did recently), so I'd suggest keeping 3.6 for now.  Adding 3.10 is hopefully just a matter of adding a test task for it.

The removals of `; python_version>="3.6"` were removals of a filter to only install those deps on Python-3, not to drop Python-3.6 support.

**Comment by djmitche** [comment]
Hm, the "re-request review" button isn't working for me, but plz re-review :)

**Review by grahamalama — APPROVED** [review]

**Review by petemoore — APPROVED** [review]

**Pull request merged**